micro-benchmarks/DRB150-gpu-missinglock1-orig-no.c

```diff
@@ -0,0 +1,38 @@
+/*
+!!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
+!!! Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+!!! and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
+!!!
+!!! SPDX-License-Identifier: (BSD-3-Clause)
+!!!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!!!
+ */
+
+/*
+The distribute parallel for directive at line 27 will execute loop using multiple teams.
+The loop iterations are distributed across the teams in chunks in round robin fashion.
+The omp lock ensures that there is no data race pair.
+*/
+
+
+#include <omp.h>
+#include <stdio.h>
+
+int main(){
+
+  omp_lock_t lck;
+  int var=0,i;
+  omp_init_lock(&lck);
+
+  #pragma omp target map(tofrom:var) device(0)
+  #pragma omp teams distribute parallel for
+  for (int i=0; i<100; i++){
+    omp_set_lock(&lck);
```

Comment:
Same as 144: locking is only guaranteed for a contention group, i.e, within a team.

Reply:
Similarly, I will restrict it to 1 team. 
  ```
#pragma omp target map(tofrom:var) device(0)
  #pragma omp teams num_teams(1)
  #pragma omp distribute parallel for
  for (int i=0; i<100; i++){
    omp_set_lock(&lck);
    var++;
    omp_unset_lock(&lck);
  }
```